server/expression/statement_runner.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/analyzer"
+
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// StatementRunner is an expression that can be added to a node to grab the statement runner.
+type StatementRunner struct {
```

Comment:
(minor) I see how this works, with the GMS `analyzer.Interpreter` interface, but it was kinda odd looking at first to see this modeled as a `sql.Expression`. Not saying we need to change this right now, but wanted to point out that it's a little non-obvious at first. Might be worth a little bit of documentation explaining that in the godoc string, to save other code readers the time of tracking down usage and trying to figure it out.

Reply:
James' GMS procedure changes add a `sql.Node` version, which I'll be able to inherit in the node rather than relying on this expression to get around that functionality. This is ultimately temporary.